src/Http/Wolverine.Http.Marten/CompiledQueryWriterPolicy.cs

```diff
@@ -14,82 +15,65 @@ public class CompiledQueryWriterPolicy : IResourceWriterPolicy
 {
     public bool TryApply(HttpChain chain)
     {
-        bool ImplementsGenericInterface(Type type, Type assigned)
+        bool ImplementsGenericInterface(Type type, Type assigned, [NotNullWhen(true)] out Type[]? types)
```

Comment:
There are worlds of Reflection helpers in JasperFx.Core for this kind of thing. There's a `Closes(type)` helper for exactly this. This logic is a huge pain, so please use that